Source/Scene/Expression.js

```diff
@@ -125,6 +128,46 @@ define([
         return CesiumMath.logBase(number, 2.0);
     }
 
+    function distance(x, y) {
+        if (typeof x === 'number' && typeof y === 'number') {
+            return Math.abs(x - y);
+        } else if (x instanceof Cartesian2 && y instanceof Cartesian2) {
+            return Cartesian2.distance(x, y);
+        } else if (x instanceof Cartesian3 && y instanceof Cartesian3) {
+            return Cartesian3.distance(x, y);
+        } else if (x instanceof Cartesian4 && y instanceof Cartesian4) {
+            return Cartesian4.distance(x, y);
+        } else {
+            //>>includeStart('debug', pragmas.debug);
+            throw new DeveloperError('Function distance requires matching types of inputs. Arguments are ' + x + ' and ' + y + '.');
+            //>>includeEnd('debug');
+        }
+    }
+
+    function dot(x, y) {
+        if (typeof x === 'number' && typeof y === 'number') {
+            return x * y;
+        } else if (x instanceof Cartesian2 && y instanceof Cartesian2) {
+            return Cartesian2.dot(x ,y);
+        } else if (x instanceof Cartesian3 && y instanceof Cartesian3) {
+            return Cartesian3.dot(x, y);
+        } else if (x instanceof Cartesian4 && y instanceof Cartesian4) {
+            return Cartesian4.dot(x, y);
+        } else {
+            //>>includeStart('debug', pragmas.debug);
+            throw new DeveloperError('Function dot requires matching types of inputs. Arguments are ' + x + ' and ' + y + '.');
+            //>>includeEnd('debug');
+        }
+    }
+
+    function cross(x, y) {
+        if (x instanceof Cartesian3 && y instanceof Cartesian3) {
+            return Cartesian3.cross(x, y, ScratchStorage.getCartesian3());
+        } else {
+            throw new DeveloperError('Invalid argument type for cross() function, must be vec3');
```

Comment:
Wrap this with `//>>includeStart('debug', pragmas.debug);` and `//>>includeEnd('debug');` too